ion-transformation-storage-api/src/main/java/com/connexta/transformation/storage/api/StorageProvider.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.transformation.storage.api;
+
+public interface StorageProvider {
+
+  /**
+   * Saves the given intel in persistence.
+   *
+   * @param intel the intel to save
+   * @throws StorageException if an error occurs while saving
+   */
+  void save(Intel intel) throws StorageException;
+
+  /**
+   * Gets the status of a transformation.
+   *
+   * @param transformId The ID of the transformation to get the status of
+   * @return the status of the transformation with the given ID
+   * @throws StorageException if an error occurs while fetching the status
+   * @throws NotFoundException if nothing associated with the given ID can be found
+   */
+  TransformInfo getStatus(String transformId) throws StorageException;
+
+  /**
+   * Gets the piece of intel with the given ID and type.
+   *
+   * @param transformId the transformation ID associated with the intel
+   * @param type the type of the intel (e.g. resource, metacard XML, IRM...)
+   * @return the {@link Intel} with the given ID and type
+   * @throws StorageException if an error occurs while fetching the intel
+   * @throws NotFoundException if nothing with the given type and ID can be found
+   */
+  Intel get(String transformId, String type) throws StorageException;
+
+  /**
+   * Deletes everything stored that is associated with the given transformation ID.
+   *
+   * @param transformId the transformation ID for which to delete all associated files
+   * @return true if the associated items were deleted successfully, false otherwise
+   * @throws NotFoundException if nothing associated with the given ID can be found
+   */
+  boolean delete(String transformId) throws StorageException;
```

Comment:
❓ If we cannot delete because of some error, wouldn't we throw back a storage exception instead? This would make this method consistent with the other ones. Not sure we need the boolean as a return value.